pkg/cmd/serve.go

```diff
@@ -26,12 +26,17 @@ func RegisterServeFlags(cmd *cobra.Command, config *server.Config) {
 	// Flags for the datastore
 	datastore.RegisterDatastoreFlags(cmd, &config.DatastoreConfig)
 
-	// Flags for the namespace manager
+	// Flags for the namespace cache
 	cmd.Flags().Duration("ns-cache-expiration", 1*time.Minute, "amount of time a namespace entry should remain cached")
 	if err := cmd.Flags().MarkHidden("ns-cache-expiration"); err != nil {
 		panic("failed to mark flag hidden: " + err.Error())
 	}
-	server.RegisterCacheConfigFlags(cmd.Flags(), &config.NamespaceCacheConfig, "ns-cache")
+	server.RegisterCacheFlags(cmd.Flags(), "ns-cache", &config.NamespaceCacheConfig, &server.CacheConfig{
+		Enabled:     true,
+		Metrics:     false,
+		NumCounters: 1_000,
+		MaxCost:     "16MiB",
+	})
```

Comment:
it wasn't immediately clear this wasn't a hardcoded configuration but a default and that CLI args take precedence. Perhaps extracting these into a variable named `namespaceCacheDefaults` would make it clearer?